app/domain/authentication/authn_azure/validate_resource_restrictions.rb

```diff
@@ -3,124 +3,93 @@
 module Authentication
   module AuthnAzure
 
-    ValidateApplicationIdentity = CommandClass.new(
+    ValidateResourceRestrictions = CommandClass.new(
       dependencies: {
-        role_class:                 ::Role,
-        resource_class:             ::Resource,
-        validate_azure_annotations: ValidateAzureAnnotations.new,
-        application_identity_class: ApplicationIdentity,
-        logger:                     Rails.logger
+        role_class:                  ::Role,
+        resource_class:              ::Resource,
+        resource_restrictions_class: ResourceRestrictions,
+        logger:                      Rails.logger
       },
       inputs:       %i(account service_id username xms_mirid_token_field oid_token_field)
     ) do
 
       def call
-        parse_xms_mirid
-        token_identity_from_claims
-        validate_azure_annotations_are_permitted
-        extract_application_identity_from_role
-        validate_required_constraints_exist
-        validate_constraint_combinations
-        validate_token_identity_matches_annotations
+        extract_resource_restrictions_from_role
+        validate_resource_restrictions_configuration
+        validate_resource_restrictions_match_request
       end
 
       private
 
-      def parse_xms_mirid
-        xms_mirid
+      def extract_resource_restrictions_from_role
+        resource_restrictions
       end
 
-      def xms_mirid
-        @xms_mirid ||= XmsMirid.new(@xms_mirid_token_field)
+      def resource_restrictions
+        @resource_restrictions ||= @resource_restrictions_class.new(
+          role_annotations: role_annotations,
+          service_id:       @service_id,
+          logger:           @logger
+        )
+      end
+
+      def validate_resource_restrictions_configuration
+        resource_restrictions.validate_configuration
```

Comment:
What is your motivation in placing the logic in ResourceRestrictions and not in ValidateResourceRestrictions? 
The functions in ResourecRestrictions are performing validation on the xms mirid fields so stylistically wondering why they aren't in their calling class

Reply:
update after chat: moved `validate_configuration` to be part of the initializer and `match_request` to `validate_resource_restrictions`